packages/core/test/unit/LinkProcessor.test.js

```diff
@@ -0,0 +1,120 @@
+const cheerio = require('cheerio');
+const fs = require('fs-extra');
+const linkProcessor = require('../../src/parsers/linkProcessor');
+
+jest.mock('fs');
+
+const json = {
+  './index.html': '1',
+  './userGuide/index.html': '2',
+  './userGuide/raw.html': '3',
+  './images/logo.png': '4',
+  './css/main.css': '5',
+};
+
+fs.vol.fromJSON(json, './src');
+
+const mockCwf = 'Test';
+
+const mockConfig = {
+  rootPath: './src',
+  baseUrl: '',
+  ignore: [
+    '_markbind', '_site/*',
+    'lib/*', '*.json',
+    '*.md', '*.mbd',
+    '*.mbdf', '*.njk',
+    '.git/*', '*.pptx',
+    'CNAME',
+  ],
+  addressablePagesSource: [
+    'index',
+    'userGuide/index',
+  ],
+};
+
+test('Test invalid URL link ', () => {
+  const mockLink = '<a href="https://markbind.org">Test</a>';
+  const mockNode = cheerio.parseHTML(mockLink)[0];
+
+  const EXPECTED_RESULT = 'Not Intralink';
+
+  expect(linkProcessor.validateIntraLink(mockNode, mockCwf, mockConfig)).toEqual(EXPECTED_RESULT);
+});
+
+test('Test valid ".html" extension link ', () => {
+  // should be checked as file asset and page
+  const mockLink = '<a href="/index.html">Test</a>';
+  const mockNode = cheerio.parseHTML(mockLink)[0];
+
+  const EXPECTED_RESULT = 'Intralink with ".html" extension is a valid Page Source or File Asset';
+
+  expect(linkProcessor.validateIntraLink(mockNode, mockCwf, mockConfig)).toEqual(EXPECTED_RESULT);
+});
+
+test('Test invalid, non-existent ".html" extension link ', () => {
+  // should be checked as file asset and page
+  const mockLink = '<a href="/missing.html">Test</a>';
+  const mockNode = cheerio.parseHTML(mockLink)[0];
+
+  const EXPECTED_ERR_RESULT = 'Intralink with ".html" extension is neither a Page Source nor File Asset';
+
+  expect(linkProcessor.validateIntraLink(mockNode, mockCwf, mockConfig)).toEqual(EXPECTED_ERR_RESULT);
+});
+
+test('Test valid raw (not page source) ".html" extension link ', () => {
+  // should be checked as file asset and page
+  const mockLink = '<a href="/userGuide/raw.html">Test</a>';
+  const mockNode = cheerio.parseHTML(mockLink)[0];
+
+  const EXPECTED_ERR_RESULT = 'Intralink with ".html" extension is a valid Page Source or File Asset';
+
+  expect(linkProcessor.validateIntraLink(mockNode, mockCwf, mockConfig)).toEqual(EXPECTED_ERR_RESULT);
+});
+
+test('Test valid link ending with /', () => {
+  // should be checked as file asset and page
+  const mockLink = '<a href="/userGuide/">Test</a>';
+  const mockNode = cheerio.parseHTML(mockLink)[0];
+
+  const EXPECTED_RESULT = 'Intralink ending with "/" is a valid Page Source or File Asset';
+
+  expect(linkProcessor.validateIntraLink(mockNode, mockCwf, mockConfig)).toEqual(EXPECTED_RESULT);
+});
+
+test('Test valid link ending with no extension', () => {
+  // should be checked as file asset and page
```

Comment:
```suggestion
  // should be checked as page
```

Reply:
needs two more that checks as an asset
some `/xx/index.html` (missed this out in earlier review)
`/xx` (the raw file)

Reply:
Added the test!